src/parser/warrior/arms/modules/core/Overpower.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import { formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+
+import Enemies from 'parser/shared/modules/Enemies';
+import ExecuteRange from './Execute/ExecuteRange';
+import SpellUsable from '../features/SpellUsable';
+
+class OverpowerAnalyzer extends Analyzer {
+    static dependencies = {
+        executeRange: ExecuteRange,
+        enemies: Enemies,
+        spellUsable: SpellUsable,
+    }
+
+    proc = 0;
+    wastedProc = 0;
+
+    hasEP = false;
+
+    constructor(...args) {
+        super(...args);
+        this.hasEP = this.selectedCombatant.hasTrait(SPELLS.EXECUTIONERS_PRECISION_TRAIT.id);
+    }
+
+    on_byPlayer_cast(event) {
+        if (event.ability.guid !== SPELLS.OVERPOWER.id) {
+            return;
+        }
+
+        this.proc += 1;
+        const overpower = this.selectedCombatant.getBuff(SPELLS.OVERPOWER.id);
+        if (!this.executeRange.isTargetInExecuteRange(event) && !this.hasEP) {
+            if (overpower && overpower.stacks === 2 && this.spellUsable.isAvailable(SPELLS.MORTAL_STRIKE.id)) {
```

Comment:
we generally try to avoid nesting ifs because it can make the branching hard to follow. you can restructure this nested-if into an early return (using the inner condition, which is shared across both branches) and then an if/elseif